### DISCUSSION TITLE
Live Map with Roborock integration rather than Xiaomi

### DISCUSSION THREAD

**Discussion by bkr1969** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#708] [Q&A] [answered]
How do I display the Roborock map if I'm not using the cloud extractor?  I just switched from the Xiaomi integration because I thought the live map was able to be displayed now.  Am I wrong?  I can't figure out out how to configure the map card if I'm not using a camera image.

**Comment by PiotrMachowski** [comment]
You can use this integration
https://github.com/humbertogontijo/homeassistant-roborock

**Reply by bkr1969** [reply]
Um...there's no Roborock integration in HACS.  There is a built in one, but there is no camera entity.

**Reply by PiotrMachowski** [reply]
You have to add it manually:
https://github.com/humbertogontijo/homeassistant-roborock?tab=readme-ov-file#installing-via-hacs

**Reply by bkr1969** [reply]
That page describes installing via HACS, not manually installing. Step one is finding it in HACS, which doesn't exist.  Am I just supposed to create a folder in my custom components and copy the files?  If I do that, won't it conflict with the already installed Roborock integration?

I'm having a hard time understanding why it's such a hassle to display a simple vacuum map.  If it can be done via the Roborock integration, then will someone please post the code?  If it can only be done via Xiaomi, why are there people discussing displaying it with Roborock?

**Reply by PiotrMachowski** [reply]
No, searching is a third step. It will appear in search results after following 2 first steps

**Reply by PiotrMachowski** [reply]
Just click this link:
https://my.home-assistant.io/redirect/hacs_repository/?owner=humbertogontijo&repository=homeassistant-roborock&category=integration

**Reply by bkr1969** [reply, marked as answer]
OK. So the map works now, but where did all my sensor info go?  There is nothing to indicate battery level or cleaning/charging status that I can see.

**Reply by bkr1969** [reply]
OK.  The sensors just converted to state attributes.  Everything seems fine now.  Thank for the help!

**Reply by sodu18** [reply]
like i understood its possible to use vacuum map card together with roborock app? because now with basic roboroc app its useless cant mark zones were to clean cant chose room...

**Reply by PiotrMachowski** [reply]
@sodu18 Yes, it is

**Reply by sodu18** [reply]
dam looks good only missing one thing can i add somewhere yml code so i would have also room select?

**Reply by PiotrMachowski** [reply]
Open card editor and click "Generate room config" button

**Reply by sodu18** [reply]
what a shame :) thanks Piotr
